src/crypto/qat/qat_crypto_accel.cc

```diff
@@ -40,3 +40,29 @@ bool QccCryptoAccel::cbc_decrypt(unsigned char* out, const unsigned char* in, si
       const_cast<unsigned char *>(&iv[0]),
       const_cast<unsigned char *>(&key[0]), CPA_CY_SYM_CIPHER_DIRECTION_DECRYPT);
 }
+
+bool QccCryptoAccel::cbc_encrypt_batch(unsigned char* out, const unsigned char* in, size_t chunk_size, size_t total_len,
+        const unsigned char iv[][AES_256_IVSIZE],
+        const unsigned char (&key)[AES_256_KEYSIZE]) {
+  if ((total_len % AES_256_IVSIZE) != 0) {
```

Comment:
one question from me. If total_len% AES_256_IVSIZE != 0, should calculate n-1 array but not the full array

Reply:
hi @optimistyzy , the total_len% AES_256_IVSIZE must be 0. Because before RGW call crypto plugin, it has done the operation.   
This judgment is retained here in order to be consistent with other existing function implementations and may prevent misuse.

It may should add `unlikely` for this:
`if (unlikely((total_len % AES_256_IVSIZE) != 0))`